### ISSUE TITLE
When using Microsoft.Data.SqlClient an InvalidCastException will be thrown

### ISSUE THREAD

**Issue by yorek** [dotMorten/Microsoft.SqlServer.Types#51]
When using Microsoft.Data.SqlClient 2.0.1 the following exception will be generated

`[System.InvalidCastException] Unable to cast object of type 'Microsoft.SqlServer.Types.SqlHierarchyId' to type 'Microsoft.Data.SqlClient.Server.IBinarySerialize'.
`
once proper assembly redirection is in place

```
 if (assemblyName.FullName == "Microsoft.SqlServer.Types, Version=10.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91")
            {
                var a = typeof(SqlGeography).Assembly;
                _logger.Info("Rebinding to " + a.FullName);
                return a;
            }

```

**Comment by dotMorten** [comment]
Microsoft.Data.SqlClient isn't supported. Likely duplicate of: https://github.com/dotMorten/Microsoft.SqlServer.Types/issues/29

**Comment by yorek** [comment]
I understand...I've submitted a PR to make it work with Microsoft.Data.SqlClient. It probably make sense to keep support to both, even if the newer Microsoft.Data.SqlClient is the preferred library (as per https://devblogs.microsoft.com/dotnet/introducing-the-new-microsoftdatasqlclient/)

**Comment by dotMorten** [comment]
Fixed in #69
